api/src/main/java/org/openmrs/module/appointmentscheduling/api/db/hibernate/HibernateAppointmentResourceDAO.java

```diff
@@ -0,0 +1,137 @@
+package org.openmrs.module.appointmentscheduling.api.db.hibernate;
+
+import org.hibernate.Query;
+import org.openmrs.Location;
+import org.openmrs.api.db.DAOException;
+import org.openmrs.module.appointmentscheduling.AppointmentResource;
+import org.openmrs.module.appointmentscheduling.BlockExcludedDays;
+import org.openmrs.module.appointmentscheduling.api.db.AppointmentResourceDAO;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.sql.Time;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+public class HibernateAppointmentResourceDAO extends HibernateSingleClassDAO
+        implements AppointmentResourceDAO {
+    /**
+     * You must call this before using any of the data access methods, since it's not actually
+     * possible to write them all with compile-time class information.
+     *
+     * @param
+     */
+    public HibernateAppointmentResourceDAO() {
+        super(AppointmentResource.class);
+    }
+
+    @Override
+    @Transactional(readOnly = true)
+    public List<AppointmentResource> getResourceByConstraints(Date startDate, Date endDate, Time startTime, Time endTime, Location location, Date appointmentDate) throws DAOException {
+        if (appointmentDate != null && getResourceExcludedDays(appointmentDate, location).contains(new SimpleDateFormat("yyyy-MM-dd").format(appointmentDate))) {
+            throw new DAOException("This date is not allowed for booking appointments");
+        } else {
+
+            if (location != null) {
+                String stringQuery = "SELECT appointmentResource FROM AppointmentResource AS appointmentResource WHERE appointmentResource.voided = 0";
+
+                if (startDate != null)
+                    stringQuery += " AND appointmentResource.startDate=:startDate";
+                if (endDate != null)
+                    stringQuery += " AND appointmentResource.endDate=:endDate";
+                if (startTime != null)
+                    stringQuery += " AND appointmentResource.startTime=:startTime";
+                if (endTime != null)
+                    stringQuery += " AND appointmentResource.endTime=:endTime";
+                if (location != null)
+                    stringQuery += " AND appointmentResource.location=:location";
+                if (appointmentDate != null) {
+                    if (!getTime(appointmentDate).equals("00:00:00")) {
```

Comment:
I'm confused if you pass in an appointment date, it only actually uses it in the if there is a time component?  What is the use case here?

Also, we shouldn't be doing a string comparison here to test that the time component is 0... you should be able to use the Joda Time library to do this.

Reply:
The challenge i'm facing is some of these dates were actually created with type util date. Does it make sense when I change their date types to util date?